packages/arb-node-core/nodehealth/nodehealth.go

```diff
@@ -0,0 +1,269 @@
+package nodehealth
+
+import (
+	"errors"
+	"math/big"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/heptiolabs/healthcheck"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+//Configuration struct
+type configStruct struct {
+	mu                         sync.Mutex
+	pollingRate                time.Duration
+	loopDelayTimer             time.Duration
+	healthcheckRPC             string
+	openethereumHealthcheckRPC string
+	primaryHealthcheckRPC      string
+	successCode                int
+	blockDifferenceTolerance   int64
+}
+
+//Log structure for passing messages on healthChan to logger
+type Log struct {
+	Err    error
+	Sev    string
+	Var    string
+	Comp   string
+	Debug  bool
+	Config bool
+
+	ValStr    string
+	ValBigInt big.Int
```

Comment:
The edge case I was worried about was if the channel was buffering and the variable was rewritten after the Log struct was pushed to the healthcheck.  This would cause unexpected behavior I think where the newly written value would be logged versus the intended value which is why I was passing by value.